File: Core CS/Algorithms/Warmup/Plus Minus/plusMinus.kt
import java.io.*
import java.text.*
import java.util.*

fun plusMinus(arr: Array<Int>): Unit {
    println("%.6f".format(arr.filter{ it > 0}.size.toFloat() / arr.size.toFloat()))
    println("%.6f".format(arr.filter{ it < 0}.size.toFloat() / arr.size.toFloat()))
    println("%.6f".format(arr.filter{ it == 0}.size.toFloat() / arr.size.toFloat()))
}

fun main(args: Array<String>) {
    val scan = Scanner(System.`in`)

    val n = scan.nextLine().trim().toInt()

    val arr = scan.nextLine().split(' ').map{ it.trim().toInt() }.toTypedArray()

    plusMinus(arr)
    scan.close()
}

File: Core CS/Algorithms/Warmup/Plus Minus/plusMinus.scala
import java.io._
import java.text._
import scala.io

object Solution {

    def plusMinus(arr: Array[Int]) {
        println("%1.6f".format(arr.filter(_ > 0).length.toFloat / arr.length.toFloat))
        println("%1.6f".format(arr.filter(_ < 0).length.toFloat / arr.length.toFloat))
        println("%1.6f".format(arr.filter(_ == 0).length.toFloat / arr.length.toFloat))

    }

    def main(args: Array[String]) {
        val stdin = scala.io.StdIn

        val n = stdin.readLine.trim.toInt

        val arr = stdin.readLine.split(" ").map(_.trim.toInt)
        plusMinus(arr)
    }
}
